site/app/models/GradingOrder.php

```diff
@@ -170,23 +196,51 @@ public function sort($type, $direction) {
      * Given the current submitter, get the previous submitter to grade.
      * Will skip students that do not need to be graded (eg no submission)
      * @param Submitter $submitter Current grading submitter
+     * @param bool $to_ungraded If the next submitter should be a submitter that is ungraded
+     * @param string|int $component_id Component ID to check (for ungraded/itempool)
+     * @param bool $to_same_itempool If the next submitter should have the same itempool item as the current submitter for the targeted component
      * @return Submitter Previous submitter to grade
      */
-    public function getPrevSubmitter(Submitter $submitter) {
+    public function getPrevSubmitter(Submitter $submitter, bool $to_ungraded = false, $component_id = "-1", $to_same_itempool = null) {
```

Comment:
Should return types be listed for this function and the function on line 230?

Reply:
I'm not sure if I was doing it wrong, but the PHP linter didn't like when I put multiple types to be returned

Reply:
Is it possible to adjust the function such that it always returns one type?  There's a reason the linter doesn't like the possibility of returning different types and that's because if you're a programmer on the other end, using this function, it can be hard to deal with the result, especially if you are unfamiliar with the code.  If it's difficult/impossible to do without multiple return types, it wouldn't be the end of the world but it does make me a little uneasy when I look at it from a long-term maintainability point of view.